File: app/models/qdm/type_tested.rb
module QDM
  # app/models/qdm/type_tested.rb
  class TypeTested < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :isDerived
    field :type
    field :preference
    field :container
    field :requirement
    field :retentionTime
    field :rejectionCriterion
    field :handling
    field :qdmVersion, type: String, default: '4.0.0'
  end
end

File: app/assets/javascripts/TypeTested.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const TypeTestedSchema = DataElementSchema({
  isDerived: Any,
  type: Any,
  preference: Any,
  container: Any,
  requirement: Any,
  retentionTime: Any,
  rejectionCriterion: Any,
  handling: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'TypeTested' },

});

module.exports.TypeTestedSchema = TypeTestedSchema;
module.exports.TypeTested = mongoose.model('TypeTested', TypeTestedSchema);
